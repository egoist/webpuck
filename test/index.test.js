const webpuck = require('../')

test('main', () => {
  const app = webpuck()

  app.src('src/index.js')
  .dest('dist/bundle.js')
  .asset(['**/*.js', '!**/node_modules/**'], 'babel-loader')

  app.extends('uglify')

  expect(app.getConfig()).toMatchSnapshot()
})
