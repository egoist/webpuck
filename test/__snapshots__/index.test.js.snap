// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
Object {
  "entry": "src/index.js",
  "module": Object {
    "rules": Array [
      Object {
        "loader": "babel-loader",
        "options": undefined,
        "test": Object {
          "and": Array [
            /\\^\\.\\*\\\\/\\.\\*\\\\\\.js\\$/,
            /\\^\\\\!\\.\\*\\\\/node_modules\\\\/\\.\\*\\$/,
          ],
        },
      },
    ],
  },
  "output": Object {
    "filename": "bundle.js",
    "path": "dist",
    "publicPath": "/",
  },
  "plugins": Array [
    UglifyJsPlugin {
      "options": Object {
        "sourceMap": true,
      },
    },
  ],
}
`;
